File: examples/03_module/stock/stock.go
// Lean Code — BSD 3-Clause License — Vivian Voss, 2026
// Scope: Read and check stock availability.

package stock

import (
	"os"
	"strconv"
	"strings"
)

func ReadStockForProduct(product string) int {
	data, err := os.ReadFile("stock.txt")
	if err != nil {
		return 0
	}

	for _, line := range strings.Split(strings.TrimSpace(string(data)), "\n") {
		parts := strings.SplitN(line, ",", 2)
		if parts[0] == product {
			count, _ := strconv.Atoi(parts[1])
			return count
		}
	}

	return 0
}

func CheckStockAvailability(product string, quantity int) bool {
	s := ReadStockForProduct(product)
	return s >= quantity
}

File: examples/03_module/stock/stock.php
<?php
// Lean Code — BSD 3-Clause License — Vivian Voss, 2026
// Scope: Read and check stock availability.

namespace Stock;

function readStockForProduct(string $product): int
{
    if (!file_exists('stock.txt')) {
        return 0;
    }

    $lines = explode("\n", trim(file_get_contents('stock.txt')));

    foreach ($lines as $line) {
        [$name, $count] = explode(',', $line, 2);
        if ($name === $product) {
            return (int) $count;
        }
    }

    return 0;
}

function checkStockAvailability(string $product, int $quantity): bool
{
    $stock = readStockForProduct($product);
    return $stock >= $quantity;
}
